src/data-processor.ts

```diff
@@ -0,0 +1,399 @@
+                        maxRenderTarget.destroy();
+                    }
+
+                    minTexture = createTexture('calcBoundMin', width, 1, PIXELFORMAT_RGBA32F);
+                    maxTexture = createTexture('calcBoundMax', width, 1, PIXELFORMAT_RGBA32F);
+
+                    renderTarget = new RenderTarget({
+                        colorBuffers: [minTexture, maxTexture],
+                        depth: false
+                    });
+
+                    maxRenderTarget = new RenderTarget({
+                        colorBuffer: maxTexture,
+                        depth: false
+                    });
+
+                    minRenderTarget = new RenderTarget({
+                        colorBuffer: minTexture,
+                        depth: false
+                    });
+
+                    minData = new Float32Array(width * 4);
+                    maxData = new Float32Array(width * 4);
+                }
+
+                return { shader, minTexture, maxTexture, renderTarget, minRenderTarget, maxRenderTarget, minData, maxData };
+            };
+        })();
+
+        // calc position
+
+        this.getPositionResources = (() => {
+            let shader: Shader = null;
+            let texture: Texture = null;
+            let renderTarget: RenderTarget = null;
+            let data: Float32Array = null;
+
+            return (width: number, height: number, numSplats: number) => {
+                if (!shader) {
+                    shader = createShaderFromCode(device, positionVS, positionFS, 'calcPositionShader', {
+                        vertex_position: SEMANTIC_POSITION
+                    });
+                }
+
+                if (!texture || texture.width !== width || texture.height !== height) {
+                    if (texture) {
+                        texture.destroy();
+                        renderTarget.destroy();
+                    }
+
+                    texture = createTexture('positionTex', width, height, PIXELFORMAT_RGBA32F);
+                    renderTarget = new RenderTarget({
+                        colorBuffer: texture,
+                        depth: false
+                    });
+                    data = new Float32Array(width * height * 4);
+                }
+
+                return { shader, texture, renderTarget, data };
+            };
+        })();
+    }
+
+    // calculate the intersection of a mask canvas with splat centers
+    intersect(options: MaskOptions | RectOptions | SphereOptions, splat: Splat) {
+        const { device } = this;
+        const { scope } = device;
+
+        const numSplats = splat.splatData.numSplats;
+        const transformA = splat.entity.gsplat.instance.splat.transformATexture;
+        const splatTransform = splat.transformTexture;
+        const transformPalette = splat.transformPalette.texture;
+
+        // update view projection matrix
+        const camera = splat.scene.camera.entity.camera.camera;
```

Comment:
Do you need the final `.camera` here?

Reply:
What is the semantic here?
splat -> the current
scene -> its scene
camera -> teh camera (so no more than one camera is supported in general? Orthographic projections could be helpful)
enity -> get the object properies of the camera (like transforms)
camera ?
camera ??

Reply:
> Do you need the final .camera here?

Nope, removed, thanks.

> What is the semantic here?

Yep, bit of a mouthful:
- splat - the SS splat instance
- scene - the SS scene instance
- camera - the SS camera
- entity - the playcanvas engine `Entity` instance containing transforms and scene/render hierarchy
- camera - the entity's camera component 
- camera (playcanvas engine'c camera instance, but component is enough so unneeded, removed)